bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/part/MultiPageEditorPart.java

```diff
@@ -291,6 +317,31 @@ protected CTabFolder createContainer(Composite parent) {
 		return newContainer;
 	}
 
+	/**
+	 * Determines the preferred tab style based on user preferences.
+	 * <p>
+	 * This method retrieves the user preference for aligning multi-page editor tabs
+	 * on top or bottom, and returns the corresponding SWT style constant.
+	 * </p>
+	 *
+	 * @return {@code SWT.TOP} if the user prefers tabs to be aligned on top,
+	 *         {@code SWT.BOTTOM} if the user prefers tabs to be aligned on the
+	 *         bottom.
+	 */
+	private int getPreferredTabStyle() {
```

Comment:
I **believe** this should be made protected and allow concrete editor implementations to override it.

Reply:
```suggestion
	protected int getPreferredTabStyle() {
```